src/model/delta/insertdelta.js

```diff
@@ -95,7 +104,44 @@ register( 'insert', function( position, nodes ) {
 	delta.addOperation( insert );
 	this.document.applyOperation( insert );
 
+	// When element is a DocumentFragment we need to move its markers to Document#markers.
+	if ( nodes instanceof DocumentFragment ) {
+		for ( const marker of nodes.markers ) {
```

Comment:
I this case destructuring might be fine (and nice) `for ( const [ markerName, markerRange ] of nodes.markers )`.